dds_web/api/project.py

```diff
@@ -188,6 +188,107 @@ def post(self):
+        if project.busy:
+            raise ProjectBusyError(
+                message=(
+                    f"The deadline for the project '{project_id}' is already in the process of being changed. "
+                    "Please try again later. \n\nIf you know the project is not busy, contact support."
+                )
+            )
+
+        self.set_busy(project=project, busy=True)
+
+        try:
+            extend_deadline = json_input.get("extend_deadline", False)  # False by default
+
+            # some variable definition
+            curr_date = dds_web.utils.current_time()
+            send_email = False
+
+            # Update the deadline
+            if extend_deadline:
+                # deadline can only be extended from Available
+                if not project.current_status == "Available":
+                    raise DDSArgumentError(
+                        "you can only extend the deadline for a project that has the status Available."
+                    )
+
+                new_deadline_in = json_input.get("new_deadline_in")
+                current_deadline = (project.current_deadline - curr_date).days
+
+                if not new_deadline_in:
+                    raise DDSArgumentError(
+                        message="No new deadline provived, cannot perform operation."
+                    )
+                if current_deadline > 10:
+                    raise DDSArgumentError(
+                        message=f"There are still {current_deadline} days left, it is not possible to extend deadline yet."
+                    )
+                if new_deadline_in + current_deadline > 90:
+                    raise DDSArgumentError(
+                        message="The new deadline needs to be less than (or equal to) 90 days."
+                    )
+
+                try:
+                    # add a fake archived status to mimick a re-release in order to have an udpated deadline
+                    new_status_row = self.expire_project(
+                        project=project,
+                        current_time=curr_date,
+                        deadline_in=project.responsible_unit.days_in_expired,
+                    )
+                    project.project_statuses.append(new_status_row)
+                    db.session.commit()
+
+                    new_status_row = self.release_project(
+                        project=project,
+                        current_time=curr_date,
+                        deadline_in=new_deadline_in + current_deadline,
+                    )
+                    project.project_statuses.append(new_status_row)
+
+                    project.busy = False  # return to not busy
+                    db.session.commit()
+
+                except (sqlalchemy.exc.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
+                    flask.current_app.logger.exception("Failed to extend deadline")
+                    db.session.rollback()
+                    raise err
```

Comment:
Should just need `raise` 
```suggestion
                    raise
```